### PR TITLE
Updated userguide to highlight the usage of automatic deploy and delete.

### PR DESCRIPTION
**Pull request by gyliu513**
/cc @marun @font

### CODE REVIEW

File: docs/userguide.md [hunk 1]
```diff
@@ -86,13 +86,16 @@ federation-v2 control plane.
...
-automated, then invoke the deployment script by running:
+Assume you are runnig the autoamted deloyment in cluster1. If you would like
+to have the deployment of the federation-v2 control plane automated, then invoke
+the deployment script by running:
 
 ```bash
 ./scripts/deploy-federation-latest.sh cluster2
 ```
 
+The above command will add both cluster1 and cluster2 to the federation control panel.
```
Comment (font) [inline docs/userguide.md]: I would mention "join" instead of "add" just to match the `kubefed2` command.

File: docs/userguide.md [hunk 2]
```diff
@@ -299,9 +302,11 @@ kubectl delete federatednamespaceplacement test-namespace
 
 ### Deployment Cleanup
 
-Run the following command to perform a cleanup of the cluster registry and
-federation deployments:
+Assume you are runnig the deloyment cleanup in cluster1. Run the following command to perform
```
Comment (font) [inline docs/userguide.md]: Same comment as above. I'd reword this. s/runnig/running/

File: docs/userguide.md [hunk 3]
```diff
@@ -299,9 +302,11 @@ kubectl delete federatednamespaceplacement test-namespace
...
+a cleanup of the cluster registry and federation deployments:
 
 ```bash
 ./scripts/delete-federation.sh cluster2
 ```
+
+The above command will delete both cluster1 and cluster2 from the federation control panel.
```
Comment (font) [inline docs/userguide.md]: I would replace "delete" with "unjoin" just to match the `kubefed2` command or provide one in parenthesis e.g. delete (unjoin).

File: docs/userguide.md (hunk 1 above)
Comment (gyliu513) [reply docs/userguide.md]: Done

File: docs/userguide.md (hunk 2 above)
Comment (gyliu513) [reply docs/userguide.md]: I deleted this assume, as the pre-condition already mean we are using `cluster 1`

File: docs/userguide.md (hunk 3 above)
Comment (gyliu513) [reply docs/userguide.md]: done

File: docs/userguide.md [hunk 4]
```diff
@@ -93,6 +93,9 @@ automated, then invoke the deployment script by running:
 ./scripts/deploy-federation-latest.sh cluster2
 ```
 
+The above script will join the host cluster to federation control panel automatically.
```
Comment (irfanurrehman) [inline docs/userguide.md]: s/panel/plane/

Comment (irfanurrehman) [reply docs/userguide.md]: Can be modified to:
The above script joins the host cluster to the federation control plane it deploys, by default.

File: docs/userguide.md [hunk 5]
```diff
@@ -93,6 +93,9 @@ automated, then invoke the deployment script by running:
...
+In above case, it means join both cluster1 and cluster2 to the federation control panel.
```
Comment (irfanurrehman) [inline docs/userguide.md]: Can be modified to:
The argument used is the context name of the additional clusters that needs to be joined to this federation control plane.
Clarifying, say the `host-cluster-context` used is `cluster1`, then on successful completion of the script used in example, both `cluster1` and `cluster2` will be joined to the deployed federation control plane.

File: docs/userguide.md [hunk 6]
```diff
@@ -305,3 +308,6 @@ federation deployments:
 ```bash
 ./scripts/delete-federation.sh cluster2
 ```
+
+The above script will unjoin the host cluster from federation control panel automatically.
```
Comment (irfanurrehman) [inline docs/userguide.md]: as above.

**Comment by irfanurrehman** [comment]
I am not sure, if this change was  necessary, as the help seems clear to me. But thinking from the point of the view of the user a double clarification does not harm.
@gyliu513 please check my comments.

File: docs/userguide.md (hunk 4 above)
Comment (gyliu513) [reply docs/userguide.md]: Done

File: docs/userguide.md (hunk 5 above)
Comment (gyliu513) [reply docs/userguide.md]: Done

File: docs/userguide.md (hunk 6 above)
Comment (gyliu513) [reply docs/userguide.md]: Done

**Comment by gyliu513** [comment]
Thanks @irfanurrehman , I think we need clarify that the host cluster will be added/removed automatically at least, and thanks for your detailed comments, all addressed now.

File: docs/userguide.md [hunk 7]
```diff
@@ -93,6 +93,12 @@ automated, then invoke the deployment script by running:
 ./scripts/deploy-federation-latest.sh cluster2
 ```
 
+The above script joins the host cluster to the federation control plane it deploys, by default.
+The argument used is the context name of the additional clusters that needs to be joined to this
```
Comment (irfanurrehman) [inline docs/userguide.md]: @gyliu513 apologies, While reading it before approving; I found the english mistake.
This should be updated as:
`The argument(s) used is/are the list of context names of the...`.
Please update and I will merge, unless @font has any further comments.

File: docs/userguide.md [hunk 8]
```diff
@@ -305,3 +311,12 @@ federation deployments:
 ```bash
 ./scripts/delete-federation.sh cluster2
 ```
+
+The above script will unjoin the host cluster from federation control panel automatically.
+In above case, it means unjoin both cluster1 and cluster2 from the federation control panel.
+
+The above script unjoins the host cluster from the federation control plane it deploys, by default.
+The argument used is the context name of the additional clusters that needs to be unjoined from
```
Comment (irfanurrehman) [inline docs/userguide.md]: same as above.

File: docs/userguide.md (hunk 7 above)
Comment (gyliu513) [reply docs/userguide.md]: Done

File: docs/userguide.md [hunk 9]
```diff
@@ -305,3 +311,12 @@ federation deployments:
 ```bash
 ./scripts/delete-federation.sh cluster2
 ```
+
+The above script will unjoin the host cluster from federation control panel automatically.
```
Comment (irfanurrehman) [inline docs/userguide.md]: Why do these two lines (315, 316) remain, when you have added the text starting from line 318 ?

Comment (gyliu513) [reply docs/userguide.md]: Fixed, thanks @irfanurrehman
